src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1825,34 +1825,36 @@ protected override void AfterMembersCompletedChecks(BindingDiagnosticBag diagnos
                         reported_ERR_InlineArrayUnsupportedElementFieldModifier = true;
                     }
 
-                    if (TryGetInlineArrayElementField() is { TypeWithAnnotations: var elementType })
-                    {
-                        NamedTypeSymbol? index = null;
-                        NamedTypeSymbol? range = null;
+                    NamedTypeSymbol? index = null;
+                    NamedTypeSymbol? range = null;
 
-                        foreach (PropertySymbol indexer in Indexers)
+                    foreach (PropertySymbol indexer in Indexers)
+                    {
+                        if (indexer.Parameters is [{ Type: { } type }] &&
+                            (type.SpecialType == SpecialType.System_Int32 ||
+                                type.Equals(index ??= DeclaringCompilation.GetWellKnownType(WellKnownType.System_Index), TypeCompareKind.AllIgnoreOptions) ||
+                                type.Equals(range ??= DeclaringCompilation.GetWellKnownType(WellKnownType.System_Range), TypeCompareKind.AllIgnoreOptions)))
                         {
-                            if (indexer.Parameters is [{ Type: { } type }] &&
-                                (type.SpecialType == SpecialType.System_Int32 ||
-                                 type.Equals(index ??= DeclaringCompilation.GetWellKnownType(WellKnownType.System_Index), TypeCompareKind.AllIgnoreOptions) ||
-                                 type.Equals(range ??= DeclaringCompilation.GetWellKnownType(WellKnownType.System_Range), TypeCompareKind.AllIgnoreOptions)))
-                            {
-                                diagnostics.Add(ErrorCode.WRN_InlineArrayIndexerNotUsed, indexer.TryGetFirstLocation() ?? GetFirstLocation());
-                            }
+                            diagnostics.Add(ErrorCode.WRN_InlineArrayIndexerNotUsed, indexer.TryGetFirstLocation() ?? GetFirstLocation());
                         }
+                    }
 
-                        foreach (var slice in GetMembers(WellKnownMemberNames.SliceMethodName).OfType<MethodSymbol>())
+                    foreach (var slice in GetMembers(WellKnownMemberNames.SliceMethodName).OfType<MethodSymbol>())
+                    {
+                        if (Binder.MethodHasValidSliceSignature(slice))
                         {
-                            if (Binder.MethodHasValidSliceSignature(slice))
-                            {
-                                diagnostics.Add(ErrorCode.WRN_InlineArraySliceNotUsed, slice.TryGetFirstLocation() ?? GetFirstLocation());
-                                break;
-                            }
+                            diagnostics.Add(ErrorCode.WRN_InlineArraySliceNotUsed, slice.TryGetFirstLocation() ?? GetFirstLocation());
```

Comment:
Btw, should we have similar warnings for user-defined `GetEnumerable()` methods?

For example, the following warns for Slice but not for GetEnumerable even though neither of them is used (and both are used if InlineArray attribute is removed):

```cs
[System.Runtime.CompilerServices.InlineArray(4)]
struct Buffer4
{
    private int _element0;

    int Length => 4;
    string Slice(int i, int j) => "int";

    public IEnumerator<char> GetEnumerator()
    {
        yield return 'e';
        yield return 'n';
        yield return 'u';
    }

    static void Main()
    {
        Buffer4 b = default;
        System.Console.WriteLine(b[..][0]);
        
        foreach (var ch in b)
        {
            System.Console.WriteLine(ch);
        }
    }
}
```

Reply:
> Btw, should we have similar warnings for user-defined GetEnumerable() methods?

There were no requests for that. Besides, the method can still be called explicitly, but there is no way to do that with an indexer or a conversion operator.